### ISSUE TITLE
When will the EAGLE3 support for Qwen2

### ISSUE THREAD

**Issue by garycaokai** [SafeAILab/EAGLE#198]
I'm impressed with Eagle3's inference performance. However, I noticed that it doesn't support Qwen2 yet. When will Qwen2 support be available?

**Comment by chtaihei-ust-hk** [comment]
Also Qwen3

**Comment by jiahe7ay** [comment]
We have successfully trained the Eagle3 versions of Qwen3-8B and Qwen3-30B-A3B based on the official training code, and have open-sourced them. On a single H200 GPU using the sglang inference framework, Qwen3-8B with Eagle3 achieves a performance boost from 186 tokens/second to 365 tokens/second, while Qwen3-30B-A3B with Eagle3 improves from 147 tokens/second to 231 tokens/second.

We used the ultra_200k test set and re-ran inference on Qwen3 to regenerate the data, which was then used as the final training set.A total of 600K dialogues were used as the training set.

https://huggingface.co/Tengyunw/qwen3_30b_moe_eagle3

https://huggingface.co/Tengyunw/qwen3_8b_eagle3

Additionally, we have also published a report detailing how to reproduce the Eagle3 training process. The report link is provided below for your reference if needed.

https://mp.weixin.qq.com/s/Dmdg6aLgFHZEcm6TY1vKkA

https://zhuanlan.zhihu.com/p/1923763301432662012

**Comment by jiahe7ay** [comment]
@garycaokai @luoruijie @chtaihei-ust-hk
